dlc-node/src/main/scala/org/bitcoins/dlc/node/DLCNode.scala

```diff
@@ -0,0 +1,60 @@
+package org.bitcoins.dlc.node
+
+import akka.actor.{ActorRef, ActorSystem}
+import grizzled.slf4j.Logging
+import org.bitcoins.core.api.dlc.wallet.DLCWalletApi
+import org.bitcoins.core.protocol.tlv._
+import org.bitcoins.core.util.StartStopAsync
+import org.bitcoins.dlc.node.config._
+import org.bitcoins.dlc.node.peer.Peer
+
+import java.net.InetSocketAddress
+import scala.concurrent._
+
+case class DLCNode(wallet: DLCWalletApi)(implicit
+    system: ActorSystem,
+    val config: DLCNodeAppConfig)
+    extends StartStopAsync[Unit]
+    with Logging {
+
+  implicit val ec: ExecutionContextExecutor = system.dispatcher
+
+  private[node] lazy val serverBindF: Future[InetSocketAddress] = {
+    logger.info(
+      s"Binding server to ${config.listenAddress}, with tor hidden service: ${config.torParams.isDefined}")
+
+    DLCServer.bind(
+      wallet,
+      config.listenAddress,
+      config.torParams
+    )
+  }
+
+  override def start(): Future[Unit] = {
+    serverBindF.map(_ => ())
+  }
+
+  override def stop(): Future[Unit] = {
+    Future.unit
```

Comment:
Probably should have some logic to kill the binding? If we don't want to address on this PR, let's file an issue so we remember to do it in the future

Reply:
I am unsure how to do this, maybe @rorp does

Reply:
You can just stop the actor.

Reply:
Or better, send `Tcp.Unbind`

Reply:
Done